### DISCUSSION TITLE
char bounding box

### DISCUSSION THREAD

**Discussion by hhslepicka** [jsvine/pdfplumber#559] [Ask for help with specific PDFs]
Hi,

I am trying to fetch each char's bounding box and compare that against the result generated by Apache PDFBox.
My goal is to reach the `cyan` box surrounding the `g` and the `A` displayed below:
![image](https://user-images.githubusercontent.com/8185425/145132899-c657e03a-5577-4532-9f12-9eaa4ab68bfd.png)

Using pdfplumber I was able to fetch the information but it seems like the box is being affected by the other chars on the same text line.
![image](https://user-images.githubusercontent.com/8185425/145133209-ba92c677-b852-48e8-ac0c-875f47257348.png)

My code with pdfplumber:
```python
import pdfplumber
file = "./teste_final.pdf"
pdf = pdfplumber.open(file)
p0 = pdf.pages[0]
p0_im = p0.to_image()
p0_im.reset()
p0_im.draw_rects([p0.chars[-13], p0.chars[-1]])
```

The code from Apache PDFBox which generated the first image is [located here](https://svn.apache.org/viewvc/pdfbox/trunk/examples/src/main/java/org/apache/pdfbox/examples/util/DrawPrintTextLocations.java?view=markup).

Here is my test PDF file:
https://drive.google.com/file/d/16V0BzcwP50f-iWhR-eRfGkZ5_o83BD6a/view?usp=sharing

Here is the full image from the first crop
![teste_final-marked-1](https://user-images.githubusercontent.com/8185425/145133515-5b7435ba-c50d-45e3-acc7-7a1ae8f9b4f4.png)
:

Thank you for your help and developing pdfplumber.

**Comment by jsvine** [comment]
Very interesting test cases you have here, @hhslepicka, and thank you for sharing! 

Practically speaking, this question would be better suited for the [`pdfminer.six`](https://github.com/pdfminer/pdfminer.six) repository; that's the dependency we use to extract character-location information. I'm unfortunately not deeply-familiar enough with the specifics of their character-bounding-box algorithm to be able to diagnose the situation myself. 

But I do wonder: What bounding boxes do you get when you remove the `g` from the `A a g` line? Does the bounding box for `A` still descend below the baseline?

**Comment by hhslepicka** [comment]
Thank you, @jsvine.
I was wondering the same so I tried with a new PDF to try and answer this question.

Here is the debug with pdfplumber/pdfminer.six:
![image](https://user-images.githubusercontent.com/8185425/145440446-451f7486-a62e-4a21-8fee-6f46ff674692.png)

Here is the debug with PDFBox:
![teste_final2-marked-1](https://user-images.githubusercontent.com/8185425/145440001-8745468f-2978-4dc2-9ac8-73d09e99a68b.png)

I will try to reach out to the pdfminer.six team as well. If I get any input I will link it here.

Thanks again for your help.

**Reply by jsvine** [reply]
Thanks for sharing that. That does seem to be the output I'd expect from `pdfminer.six`. And in the case of `pdfplumber`, I think those are indeed the bounding boxes we typically want — especially for the purpose of grouping characters on the same line. That said, I could definitely see it being useful if `pdfminer.six` provided users with the option for closer-cropped bounding boxes. Curious to hear what you learn!
